### PR TITLE
Add support for bus_type nvme

### PR DESCRIPTION
**Pull request by Gnitset**
We're using nvme for our disks and need support to resize them in our deployment.

Right not that is not supported, but this fix should make it work.
It works for us.

### CODE REVIEW

**Review by Didainius — COMMENTED** [review]
Thanks for this PR. 
A few more queries:
* Would you be able to add a CHANGELOG note - create a file in `.changes/v3.3.0` directory named 68-improvements.md and add a line as in similar comments
* Add nvme in docs `vapp_vm.html.markdown` and `vm_internal_disk.html.markdown` files

**Comment by Didainius** [comment]
@Gnitset , 
Thanks for adjustments. I just made a PR for your PR branch (hope this is not too complicated). 
The main reason is we just want to have a test to cover that functionality and I am not sure you have a test env.

https://github.com/Gnitset/terraform-provider-vcd/pull/1

I should be good to go after that, we'll just ask a second look from @dataclouder afterwards

**Review by dataclouder — APPROVED** [review]

File: .changes/v3.3.0/680-improvements.md
Comment (Didainius) [inline .changes/v3.3.0/680-improvements.md:1]: Sorry for having this stuck. Could you just merge with master and change .changes to `v3.4.0`. I will just double check after latest merge with master and approve it

Comment (Didainius) [reply .changes/v3.3.0/680-improvements.md:1]: ping

Comment (Gnitset) [reply .changes/v3.3.0/680-improvements.md]: rebased on upstream master and moved .changes to v3.4.0

File: website/docs/r/vapp_vm.html.markdown
Comment (dataclouder) [inline website/docs/r/vapp_vm.html.markdown:363]:
```suggestion
* `bus_type` - (Required) The type of disk controller. Possible values: `ide`, `parallel`( LSI Logic Parallel SCSI), `sas`(LSI Logic SAS (SCSI)), `paravirtual`(Paravirtual (SCSI)), `sata`, `nvme`. **Note** `nvme` requires *v3.4.0+* and VCD *10.2.1+*
```

File: website/docs/r/vm_internal_disk.html.markdown
Comment (dataclouder) [inline website/docs/r/vm_internal_disk.html.markdown:46]:
```suggestion
* `bus_type` - (Required) The type of disk controller. Possible values: `ide`, `parallel`( LSI Logic Parallel SCSI), `sas`(LSI Logic SAS (SCSI)), `paravirtual`(Paravirtual (SCSI)), `sata`, `nvme`. **Note** `nvme` requires *v3.4.0+* and VCD *10.2.1+*
```

File: vcd/resource_vcd_vm_internal_disk_test.go
Comment (dataclouder) [inline vcd/resource_vcd_vm_internal_disk_test.go:387]:
```suggestion
		"FuncName": t.Name(),
```

**Review by dataclouder — CHANGES_REQUESTED** [review]
Good job!
Some small changes required

**Comment by Didainius** [comment]
Closing this one in favor of #739 which should include addresses comments.

**Pull request closed without merging**
